src/gui/Layout.cpp

```diff
@@ -434,27 +417,24 @@ XojPageView* Layout::getViewAt(int x, int y)
 
 	}
 
-
 	if (testCol < numCols && testRow < numRows)
 	{
-		int page = this->mapper.map(testCol, testRow);
+		auto optionalPage = this->mapper.map(testCol, testRow);
 		this->lastGetViewAtRow = testRow;
 		this->lastGetViewAtCol = testCol;
 
-		if (page >= 0 && this->view->viewPages[page]->containsPoint(x, y, false))
+		if (optionalPage && this->view->viewPages[*optionalPage]->containsPoint(x, y, false))
 		{
-			return this->view->viewPages[page];
+			return this->view->viewPages[*optionalPage];
 		}
-
 	}
 
-	return NULL;
+	return nullptr;
 }
 
-int Layout::getIndexAtGridMap(int row, int col)
+boost::optional<size_t> Layout::getIndexAtGridMap(size_t row, size_t col)
```

Comment:
This is why the build fails. You will have to exchange this with something else...

Reply:
Easy way out is `std::unique_ptr`. But this is temporary; using pointers for optional values is really terrible.

Reply:
replaced it with my own impl of an optional<size_t>.

> Easy way out is std::unique_ptr. But this is temporary; using pointers for optional values is really terrible.

I saw cases where the llvm/clang compiler optimized scoped std::unique_ptr completely away :D